src/main/kotlin/com/senacor/lpt/service/creditapplication/CreditApplicationsController.kt

```diff
@@ -21,23 +22,23 @@ class CreditApplicationsController(
     var logger: Logger = getLogger(CreditApplicationsController::class.java)
 
     @PostMapping
-    fun evaluateCreditApplication(@RequestBody request: CreditApplicationRequest): Mono<CreditDecision> {
-        return customerMasterDataClient.selectCustomerById("12983719")
-            .map {
-                // TODO: put fancy credit application evaluation logic here...
-                CreditDecision(CreditDecisionType.APPROVED)
-            }
-            .flatMap { creditDecision ->
+    fun evaluateCreditApplication(@RequestBody request: CreditApplicationRequest): Mono<CreditApplicationResponse> =
+        Mono.zip(Mono.just({
+            // TODO: put fancy credit application evaluation logic here...
+            CreditDecision(CreditDecisionType.APPROVED)
+        }), customerMasterDataClient.selectCustomerById("12983719"))
+            .map { creditDecisionCustomerTuple ->
                 // TODO: add a proper application/domain layer instead of just talking to a repo
-                val creditApplication = request.toDomain()
-                    .copy(creditDecision = creditDecision.decision)
-
-                creditApplicationRepository.save(toFirestoreModel(creditApplication))
-                    .map { creditDecision }
+                request.toDomain().copy(creditDecision = creditDecisionCustomerTuple.t1.invoke().decision)
+            }.flatMap { domainModel ->
+                creditApplicationRepository.save(toFirestoreModel(domainModel))
+            }.map { savedCreditApplication ->
+                CreditApplicationResponse(
+                    CreditDecision(savedCreditApplication.creditDecision), savedCreditApplication.id
+                )
             }
-    }
 
-    @PostMapping("/{id}/accept")
+    @PostMapping("/{id}")
```

Comment:
Bin zu alt für REST-Kriege, einen POST auf die Resource würde ich nicht machen. Finden die vorherige Version besser.

Reply:
In dieser REST Logik erstellt man die Ressource mit dem POST, wenn man sich dazu entscheidet den Kredit anzulegen. Ein Verb in der URL ist immer doppelt mit dem HTTP Verb. Können auch gerne zu einem PUT oder zu "acceptance" wechseln